File: Longest_Palindrome_Substring/longest_palindrome_substring.java
class Solution {
    public String longestPalindrome(String s) {
        int max_len = 0;
        String res = "";
        int s_len = s.length();
        for (int i = 0; i < s_len; i++) {
            int start = i;
            int end = i;
            while (start >= 0 && end < s_len && s.charAt(start) == s.charAt(end)) {
                start--;
                end++;
            }
            int diff = end - start + 1;
            if (diff > max_len) {
                max_len = diff;
                res = s.substring(start + 1, end);
            }
            start = i;
            end = i + 1;
            while (start >= 0 && end < s_len && s.charAt(start) == s.charAt(end)) {
                start--;
                end++;
            }
            diff = end - start + 1;
            if (diff > max_len) {
                max_len = diff;
                res = s.substring(start + 1, end);
            }
        }
        return res;
    }
}

File: Longest_Palindrome_Substring/longest_palindrome_substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if s == s[::-1]: return s
        max_len = 0
        res = ""
        for curr in range(len(s)):
            start = curr
            end = curr
            while (start >= 0 and end < len(s) and s[start] == s[end]):
                start -= 1
                end += 1
            diff = end - start + 1
            if (max_len < diff):
                max_len = diff
                res = s[start + 1:end]
            start = curr 
            end = curr + 1
            while (start >= 0 and end < len(s) and s[start] == s[end]):
                start -= 1
                end += 1
            diff = end - start + 1
            if (max_len < diff):
                max_len = diff
                res = s[start + 1:end]
        return res
    
